Revert increase in default pool size

justinstoller opened the pull request:
After updating the default pool_size, a race condition was
found when initializing our local git cache. See
#1058 for the issue in detail.

When attempting a fix we realized that there were multiple copies of the
code to iterate over modules to manage. See
#1059 for a first attempt at this
problem and discussion.

In order to meet our current timelines, the default will need to reverted
until we can fix this issue properly. Hence this commit and a55536c.

This commit was necessary because the CHANGELOG has been refactored since
the commit referencing the default was added.

mwaggett commented on spec/unit/action/puppetfile/install_spec.rb:
```diff
@@ -19,9 +19,7 @@ def installer(opts = {}, argv = [], settings = {})
-      (1..4).map do |idx|
-        R10K::Module::Base.new("author/modname#{idx}", "/some/nonexistent/path/modname#{idx}", nil)
-      end
+      Array.new(4, R10K::Module::Base.new('author/modname', "/some/nonexistent/path/modname", nil))
```
looks like we might want to keep this? https://github.com/puppetlabs/r10k/pull/1044/commits/0e198cac586cf79af5ad1ce0b1305a3eb02846e1

justinstoller replied:
Updated

justinstoller committed:
Revert "Increase default pool_size to 4"

This reverts commit f9f801bf7c87a78194b8a7c1b0c567678791ee4b.

justinstoller committed:
(maint) Remove CHANGELOG note re default thread pool size

After updating the default pool_size, a race condition was
found when initializing our local git cache. See
https://github.com/puppetlabs/r10k/issues/1058 for the issue in detail.

When attempting a fix we realized that there were multiple copies of the
code to iterate over modules to manage. See
https://github.com/puppetlabs/r10k/pull/1059 for a first attempt at this
problem and discussion.

In order to meet our current timelines, the default will need to reverted
until we can fix this issue properly. Hence this commit and a55536cad.

This commit was necessary because the CHANGELOG has been refactored since
the commit referencing the default was added.
